src/main.rs

```diff
@@ -190,15 +190,20 @@ fn serialize_dense_nodes(
                 node.set_tag_first_idx(tags.next_index());
                 loop {
                     let k = dense_nodes.keys_vals[tags_offset];
+                    tags_offset += 1;
```

Comment:
Could you replace tags_offset with an iterator? Having to call ```next``` would make this work automatically

Reply:
Good idea.

Reply:
Hmm... I tried to use the range `0..` for that but it feels unnatural to write `.next().unwrap()`. I also started to implement my own struct for that, but this seems also too much.